online-shop/src/components/Details/smartDetails.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { useState } from "react";
+import "../../productModeling/product.css";
+import {
+  IProduct,
+  ICartProduct,
+} from "../../model/Interfaces";
+import { library } from "@fortawesome/fontawesome-svg-core";
+import { faShoppingBasket, faEraser } from "@fortawesome/free-solid-svg-icons";
+import { AppState } from "../../reducers/combine";
+import { Dispatch } from "redux";
+import { connect } from "react-redux";
+import {
+  loadProduct,
+  deleteProduct,
+  showThePopUp,
+  hideThePopUp,
+  fetchSelected,
+  fetchDelete,
+  changeLoadingIndicator
+} from "../../actions/productActions";
+import { loadCart, eraseItem, switchLoading } from "../../actions/shoppingActions";
+import { loadProducts } from "../../actions/productListActions";
+import { displayProduct } from "../../actions/editActions";
+import defaultImg from "../../default.jpg";
+import { IDumbDetails, ProductDetailsView } from "./dumbDetails";
+import { number } from "prop-types";
+import { lifecycle, compose } from "recompose";
+import { withLoading } from "../HOCS/LoaderHOC";
+
+library.add(faShoppingBasket, faEraser);
+
+export interface LocalState {
+  match?: any;
+  onAddProduct?: any;
+  onDeleteProduct?: any;
+}
+
+interface ProductDetailsProps {
+  data: IProduct;
+  match?: any;
+  onAddProduct?: any;
+  onDeleteProduct?: any;
+  loadProduct: (data: IProduct) => void;
+  showModel: boolean;
+  messagePopUp: string;
+  titlePopUp: string;
+  productToDelete: IProduct;
+  addToCart: (product: IProduct) => void;
+  deleteProduct: (product: IProduct) => void;
+  fetchSelected: (id: number) => void;
+  showThePopUp: (message: string, title: string) => void;
+  hideThePopUp: () => void;
+  reloadProductList: (newList: IProduct[]) => void;
+  deleteProductFromCart: (product: IProduct) => void;
+  transferProductToEdit: (product: IProduct, msg: string) => void;
+  crtCart: ICartProduct[];
+  fetchDelete: (id: number,loadingInd:boolean)=> void;
+  isLoading:boolean;
+  switchLoading:()=>void;
+}
+const onComponentDidMount = lifecycle<ProductDetailsProps, {}, {}>({
```

Comment:
How come this is outside of the class defined below? Using `lifecycle` makes sense just when you don't have an underlying class (but a functional component - a dumb component).